raft/handler.go

```diff
@@ -69,6 +69,7 @@ func (h *Handler) serveJoin(w http.ResponseWriter, r *http.Request) {
 
 	// Return member's id in the cluster.
 	w.Header().Set("X-Raft-ID", strconv.FormatUint(id, 10))
+	w.Header().Set("X-Raft-Leader-ID", strconv.FormatUint(leaderID, 10))
```

Comment:
It was a simple change and it cleans up the trace log since it doesn't have to keep trying to reconnect while waiting for a heartbeat.